### PR TITLE
Closes #2049 Date/Time Imputation Utilities are no Longer listed on pkgdown site

### LINKED ISSUE #2049: Documentation:  Date imputation functions should have `@keywords internal` added
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

As part of our efforts to get to a `1.0` version of admiral, we are looking over all functions for consistency and user-friendliness.

Currently, these functions are displayed on the Reference page but are not accessible to users as we do not export them.  We should suppress these functions from the webpage by doing the following:

Adding `@internal` to the roxygen header to suppress building them onto the reference page.  This would still make them accessible to developers using `?function_name` if they want to read the documentation for these functions.

image

## Definition of Done

- [ ] The seven function shown above are suppressed from the Reference page (please build website locally to check)
- [ ] Double check they are not being referenced or discussed in any documentation
- [ ] Remove Utilities used for Date Imputation section from the reference page - https://github.com/pharmaverse/admiral/blob/devel/man/roxygen/meta.R

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #2049 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: man/roxygen/meta.R [hunk 1]
```diff
@@ -16,7 +16,6 @@ list(
     utils_fmt = "Utilities for Formatting Observations: ",
     utils_help = "Utilities used within Derivation functions: ",
     utils_examples = "Utilities used for examples and template scripts: ",
-    utils_impute = "Utilities used for date imputation: ",
```
Comment (bms63) [inline man/roxygen/meta.R:19]: Umm....this is for the families right?  Should we keep this if we are keeping the @families in the roxygen?

Comment (ddsjoberg) [reply man/roxygen/meta.R:19]: `utils_impute` was the keyword that was used to group these functions in the Reference page, which we no longer are using. All these non-exported functions still have the `@family utils_impute` tag, which links the help files together in the See Also section, e.g.

image

To be 100% upfront, I've never used the `meta.R` roxygen2 script. Are you seeing anything unexpected in the documentation?

Comment (ddsjoberg) [reply man/roxygen/meta.R:19]: Did more reading, and you're 100% correct about `meta.R`. Here's what it looks like with that entry back: (the family title is modified)

image

Comment (bms63) [reply man/roxygen/meta.R]: Awesome.  Yeah I set this up, but couldn't remember what our decision was around families for functions not visible on website.  I think this is nice for a developer when looking for other dtm imputation functions.

**Review by bms63 — APPROVED** [review]
Let's go!!

**Pull request merged**